File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand1, strand2)
    strand1_ary = strand1.split("")
    strand2_ary = strand2.split("")
    if strand1_ary.length != strand2_ary.length
      raise ArgumentError
    end
    counter = 0
    result = strand1_ary.zip(strand2_ary).map { |x, y| x == y }
    result.map do |element|
      if element == false
        counter += 1
      end
    end
    counter
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {

  Hamming.prototype.compute = function(strandA, strandB) {
    strandA = strandA.split('');
    strandB = strandB.split('');
    if (strandA.length != strandB.length) {
      throw Error('DNA strands must be of equal length.')
    }
    var counter = 0;
    for (var i = 0; i < strandA.length; i++) {
      if (strandA[i] != strandB[i]) {
        counter ++
      }
    }
    return counter;
  }
}

module.exports = Hamming;
